mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -727,6 +727,194 @@ public enum MremapFlags : ulong {
 		MREMAP_MAYMOVE = 0x1,
 	}
 
+	[Map]
+	[CLSCompliant (false)]
+	public enum SockType : int {
+		SOCK_STREAM    =  1, // Byte-stream socket
+		SOCK_DGRAM     =  2, // Datagram socket
+		SOCK_RAW       =  3, // Raw protocol interface (linux specific)
+		SOCK_RDM       =  4, // Reliably-delivered messages (linux specific)
+		SOCK_SEQPACKET =  5, // Sequenced-packet socket
+		SOCK_DCCP      =  6, // Datagram Congestion Control Protocol (linux specific)
+		SOCK_PACKET    = 10, // Linux specific
+	}
+
+	[Map][Flags]
+	[CLSCompliant (false)]
+	public enum SockFlags : int {
```

Comment:
This should be named `SocketFlags`...though shouldn't these values be in `SocketType`?

From the [**socket**(2) man page](http://man7.org/linux/man-pages/man2/socket.2.html):

> Since Linux 2.6.27, the type argument serves a second purpose: in
> addition to specifying a socket type, it may include the bitwise OR
> of any of the following values, to modify the behavior of socket():